app/controllers/contactNumbers/ConfirmPhoneNumbersController.scala

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.contactNumbers
+
+import common.SessionKeys
+import common.SessionKeys._
+import config.{AppConfig, ErrorHandler}
+import controllers.BaseController
+import controllers.contactNumbers.routes
+import controllers.predicates.AuthPredicateComponents
+import javax.inject.{Inject, Singleton}
+import models.customerInformation.UpdatePPOBSuccess
+import models.errors.ErrorModel
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.VatSubscriptionService
+import utils.LoggerUtil.{logInfo, logWarn}
+import views.html.contactNumbers.ConfirmPhoneNumbersView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ConfirmPhoneNumbersController @Inject()(val authComps: AuthPredicateComponents,
+                                       override val mcc: MessagesControllerComponents,
+                                       val errorHandler: ErrorHandler,
+                                       val vatSubscriptionService: VatSubscriptionService,
+                                       confirmPhoneNumbersView: ConfirmPhoneNumbersView,
+                                       implicit val appConfig: AppConfig) extends BaseController(mcc, authComps) {
+
+  implicit val ec: ExecutionContext = mcc.executionContext
+
+  def show: Action[AnyContent] = allowAgentPredicate { implicit user =>
+    val prepopulationLandline = user.session.get(prepopulationLandlineKey).filter(_.nonEmpty)
+    val validationLandline = user.session.get(validationLandlineKey).filter(_.nonEmpty)
+    val prepopulationMobile = user.session.get(prepopulationMobileKey).filter(_.nonEmpty)
+    val validationMobile = user.session.get(validationMobileKey).filter(_.nonEmpty)
+
+    def numberToShow(prepopulatedValue : Option[String], validationValue : Option[String]) : String = {
+      (prepopulatedValue, validationValue) match {
+        case (None, None) => request2Messages(user)("confirmPhoneNumbers.notProvided")
+        case (None, _) => request2Messages(user)("confirmPhoneNumbers.numberRemoved")
+        case (Some(prepop), _) => prepop
+      }
+    }
+
+    (prepopulationLandline, prepopulationMobile) match {
+      case (None, None) => Redirect(controllers.contactNumbers.routes.ConfirmPhoneNumbersController.show())
```

Comment:
This is redirecting to the same action here. Is it worth changing this, now that you have access to `CaptureContactNumbersController`? I'm aware this is maybe encroaching onto the wiring up task though.

Reply:
Yeah I'd say it's more going into the wiring up task, but I'm making changes anyway and it's only little so i'll just bundle that in